src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java

```diff
@@ -131,6 +133,82 @@ public void run() {
     }
 
     /**
+     * Look for running libsyncthing.so processes and kill them.
+     * Try a SIGTERM once, then try again (twice) with SIGKILL.
+     *
+     */
+    public static void killSyncthing() {
+        for (int i = 0; i < 2; i++) {
+            Process ps = null;
+            DataOutputStream psOut = null;
+            try {
+                ps = Runtime.getRuntime().exec("sh");
+                psOut = new DataOutputStream(ps.getOutputStream());
+                psOut.writeBytes("ps | grep libsyncthing.so\n");
+                psOut.writeBytes("exit\n");
+                psOut.flush();
+                ps.waitFor();
+                InputStreamReader isr = new InputStreamReader(ps.getInputStream());
+                BufferedReader br = new BufferedReader(isr);
+                String id;
+                while ((id = br.readLine()) != null) {
+                    killProcessId(id, i > 0);
+                }
```

Comment:
You don't seem to check if the kill succeeds, so you run it 3 times in any case. Which gives us the same problem if someone stops and immediately starts the app.